GameStore.API/Controllers/CommentsController.cs

```diff
@@ -15,20 +19,60 @@ public CommentsController(ICommentService commentService)
         _commentService = commentService;
     }
 
-    [HttpPost("game/{gamekey}/newComment")]
+    [HttpPost("game/{gameKey}/newComment")]
     public async Task<ActionResult<CommentModel>> AddComment(string gameKey, CommentCreateModel comment)
     {
-        //Later i'll extract username from JWT, but for now i left it hardcoded.
-        var created = await _commentService.AddAsync("bekirov", gameKey, comment);
+        var userName = User.GetUserName();
```

Comment:
I propose the idea of storing the user principial in a separate **scoped**(single instance per entire api request chain) service and using it in services where you need to use data from JWT. Save the principial in service before request reaches controller (in middleware).

Reply:
We just want to get rid of this line and improve scalability